src/FieldSelect.tsx

```diff
@@ -0,0 +1,322 @@
+  // Disabling the line below because of an existing bug with eslint
+  // that interprets the function below as a stateless/functional component
+  // therefore erring because propTypes are not defined
+  // ref: https://github.com/yannickcr/eslint-plugin-react/issues/1919
+
+  // eslint-disable-next-line
+    ({ field: { onBlur, value }, form: { errors, touched, setFieldValue, submitCount = 0 } }: FieldProps) => (
+    <FieldSelect
+      ariaLabel={ariaLabel}
+      ariaLabelledby={ariaLabelledby}
+      autoFocus={autoFocus}
+      className={className}
+      closeMenuOnSelect={closeMenuOnSelect}
+      disabled={disabled}
+      helpText={helpText}
+      hideLabel={hideLabel}
+      id={id}
+      isClearable={isClearable}
+      isInvalid={
+        !!(errors[name] && (isTouched(touched, name) || submitCount > 0))
+      }
+      isReadOnly={isReadOnly}
+      label={label}
+      loading={loading}
+      loadingMessage={loadingMessage}
+      menuIsOpen={menuIsOpen}
+      menuPlacement={menuPlacement}
+      menuPortalTarget={menuPortalTarget}
+      multiSelect={multiSelect}
+      noOptionsMessage={noOptionsMessage}
+      onBlur={onBlur}
+      onChange={generateOnChangeHandler({ name, setFieldValue, onChange })}
+      onFocus={onFocus}
+      options={options}
+      placeholder={placeholder}
+      required={required}
+      showCheckbox={showCheckbox}
+      size={size}
+      validationText={errors[name]}
+      value={value}
+      width={width}
+    />
+  );
+
+/**
+ * @description Adapter component that connects the Formik Field component with
+ * the pebble FieldSelect component
+ *
+ * FieldSelect ref: https://pebble.istreamplanet.net/#/Components/FieldSelect
+ */
+const FieldSelectAdapter = ({
+  ariaLabel,
+  ariaLabelledby,
+  autoFocus,
+  className,
+  closeMenuOnSelect,
+  disabled,
+  helpText,
+  hideLabel,
+  id,
+  isClearable,
+  isReadOnly,
+  label,
+  loading,
+  loadingMessage,
+  menuIsOpen,
+  menuPlacement,
+  menuPortalTarget,
+  multiSelect,
+  name,
+  noOptionsMessage,
+  onChange,
+  onFocus,
+  options,
+  placeholder,
+  required,
+  showCheckbox,
+  size,
+  type,
+  value,
+  width,
+}: FieldSelectAdapterOptions): JSX.Element => (
+  <Field name={name} type={type}>
+    {generateFieldSelectRenderProp({
+      ariaLabel,
+      ariaLabelledby,
+      autoFocus,
+      className,
+      closeMenuOnSelect,
+      disabled,
+      helpText,
+      hideLabel,
+      id,
+      isClearable,
+      isReadOnly,
+      label,
+      loading,
+      loadingMessage,
+      menuIsOpen,
+      menuPlacement,
+      menuPortalTarget,
+      multiSelect,
+      name,
+      noOptionsMessage,
+      onChange,
+      onFocus,
+      options,
+      placeholder,
+      required,
+      showCheckbox,
+      size,
+      value,
+      width,
+    })}
+  </Field>
+);
+
+interface FieldSelectAdapterOptions extends InputOptions {
+  /**
+   * The id attribute of the FieldSelect's container
+   */
+  id: string;
```

Comment:
```suggestion
  id?: string;
```
`id` is option in all the other controls.